src/gtfs/read.rs

```diff
@@ -1277,17 +1277,62 @@ where
                         .collect();
                     start_time = start_time + Time::new(0, 0, frequency.headway_secs);
                     let generated_vj = VehicleJourney {
-                        id: generated_trip_id,
+                        id: generated_trip_id.clone(),
                         service_id,
                         stop_times,
                         ..corresponding_vj.clone()
                     };
                     new_vehicle_journeys.push(generated_vj);
+                    let stop_time_ids: HashMap<(String, u32), String> = corresponding_vj
+                        .stop_times
+                        .iter()
+                        .filter(|stop_time| {
```

Comment:
When you `filter()` on a test that does

```rust
.filter(|a| first_check(a) && second_check(a))
```

It might be more readable to split it like

```rust
.filter(|a| first_check(a))
.filter(|a| second_check(a))
```

Reply:
It's not the same, i need an "or"

Reply:
`&&` is an AND, not an OR. I am not talking about this test

```rust
stop_time.pickup_type == 2 || stop_time.drop_off_type == 2 
```
but about this entire test
```rust
(stop_time.pickup_type == 2 || stop_time.drop_off_type == 2)
    && collections
        .stop_time_comments
        .get(&(frequency.trip_id.clone(), stop_time.sequence))
        .is_some()
```

Take a look at my PR to see what I mean.

Reply:
Done